doc/source/reference.rst

```diff
@@ -54,6 +59,45 @@ This also helps the story of running behind
 a corporate firewall: the `localhost` can be (connecting to)
 a bastion host with the "deployment" actions happening behind the firewall.
 
+run.sh
+------
+
+Instead of running a series of scripts, unrelated to each other, and
+hard to remember for a deployer, SOCok8s takes the approach of a
+single shell script to drive from A to Z a deployment.
+
+The single point of access brings multiple features:
+
+* It's now possible to always ensure shell environment
+  variables are always set, while keeping the functional
+  behaviour in its own dedicated shell script.
+* It's now possible to ensure `localhost` have all the
+  system requirements before deploying, without asking
+  for user intervention.
+
+run.sh is a bash script, because it is a very commonly
+installed software on the 'localhost' node, independent
+of the distribution or of its version.
+It allows to install higher level requirements,
+like ansible, until we package socok8s differently.
+
+Each of the steps in `run.sh` are written in a way they represent a
+user facing feature. While what happens behind the scenes could
+change, the user interface is, in theory, stable.
```

Comment:
It might be good to mention that some those those "user facing features" are actually intended for developers...

Reply:
A developer is a user? I am afraid if I change this it will be more confusing.

Reply:
Developer is only one kind of user. The actual end users of the product are not supposed to use developer facing commands. But anyway, you can ignore that.